contracts/Puzzle.sol

```diff
@@ -0,0 +1,69 @@
+pragma solidity >=0.4.22; 
+contract Puzzle {
+    string internal constant INSUFFICIENT_FUND_MESSAGE = "Insufficient Fund";
+    string internal constant RESTRICTED_MESSAGE = "Unauthorized Access";
+    uint amount;
+    mapping(address => uint) bestlevel;
+    address public manager;  // The adress of the owner of this contract
+    address payable[] public players;  // The players of current session
+
+    constructor() public {
+        manager = msg.sender;
+    }
+
+    /**
+     * @dev The player enters into the current game session by
+     *      paying at least 2 token.
+     */
+    function play() public payable {
+        require(msg.value == 2 ether, INSUFFICIENT_FUND_MESSAGE);
+        players.push(msg.sender);
+    }
+
+    /**
+     * @dev returns the current best level the player has crossed   
+     */
+    function getlevel() public view returns(uint) {
+        return bestlevel[msg.sender];
+    }
+
+    /**
+     * @dev returns the current best level the player has crossed   
+     */
+    function setlevel(address player, uint level) public restricted{
+       bestlevel[player] = level;
+    }
+
+    /**
+     * @dev pay the player if they have crossed their last best level. Set tjeir
+     */
+    function payout(address payable player, uint level) public payable restricted {
+        if ( level > bestlevel[player]) {
+            amount = level - bestlevel[player];
```

Comment:
amount is way too less.  should be calculated dynamically.  Please use a function to calculate the amount.

Reply:
okay!